### PR TITLE
Remove semantic convention automation from Specification repository.

### PR DESCRIPTION
**Pull request by jsuereth**
Removes semantic convention automation and programmatic pieces now that semantic conventions reside in a separate repository:

https://github.com/open-telemetry/semantic-conventions

Related [OTEP(s)](https://github.com/open-telemetry/oteps) [#0227](https://github.com/open-telemetry/oteps/blob/main/text/0227-separate-semantic-conventions.md)

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM. @jsuereth I think it would make sense to add a changelog entry for 1.21.0 release "Semantic conventions have moved from this repository to https://github.com/open-telemetry/semantic-conventions"

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -83,16 +83,3 @@ jobs:
     - name: run docfx
       run: docfx build --dry-run
```
Comment (pyohannes) [inline .github/workflows/checks.yaml:85]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -71,30 +71,14 @@ install-yamllint:
...
-
-# Check if current markdown tables differ from the ones that would be generated from YAML definitions
-.PHONY: table-check
-table-check:
-	docker run --rm -v $(PWD)/semantic_conventions:/source -v $(PWD)/specification:/spec \
-		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source markdown -md /spec --md-check
-
-.PHONY: schema-check
-schema-check:
-	$(TOOLS_DIR)/schema_check.sh
```
Comment (pyohannes) [inline Makefile:88]: I guess we can get rid of this file too (`internal/tools/schema_check.sh`)?

**Review by pyohannes — APPROVED** [review]

File: Makefile (hunk 2 above)
Comment (jsuereth) [reply Makefile:88]: Yep! done.

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
When we are ready to merge, we will need to manually disable the schemas-check/semantic-conventions checks in our main branch protection rules.

**Review by joaopgrassi — APPROVED** [review]

File: specification/overview.md [hunk 3]
```diff
@@ -72,17 +72,14 @@ Application owners use the SDK constructors; [plugin authors](glossary.md#plugin
...
-* [Metrics Conventions](metrics/semantic_conventions/README.md)
+Semantic Conventions are now located in their own repository:
+[https://github.com/open-telemetry/semantic-conventions](https://github.com/open-telemetry/semantic-conventions)
 
 Both the collector and the client libraries SHOULD autogenerate semantic
 convention keys and enum values into constants (or language idiomatic
 equivalent). Generated values shouldn't be distributed in stable packages
 until semantic conventions are stable.
-The [YAML](../semantic_conventions/README.md) files MUST be used as the
+The [YAML](https://github.com/open-telemetry/semantic-conventions) files MUST be used as the
```
Comment (arminru) [inline specification/overview.md:82]: They are in this subfolder:
```suggestion
The [YAML](https://github.com/open-telemetry/semantic-conventions/tree/main/semantic_conventions) files MUST be used as the
```

**Review by arminru — APPROVED** [review]

**Comment by carlosalberto** [comment]
We should probably do a follow-up to triple check outdated notes, but otherwise we are ready to go!

**Pull request merged**